recipe/meta.yaml

```diff
@@ -46,26 +46,34 @@ outputs:
       missing_dso_whitelist:
         - '*'
       runpath_whitelist:
-        - "x86_64-conda_cos6-linux-gnu/sysroot/lib/libpthread-2.12.2.so"
+        - "x86_64-{{ ctng_vendor }}-linux-gnu/sysroot/lib/libpthread-2.12.2.so"
+      ignore_run_exports:
+        - __glibc
     requirements:
       host:
+        - sysroot_{{ ctng_target_platform }} {{ conda_glibc_ver }}
         - {{ pin_subpackage("libgomp", exact=True) }}
       run:
         - {{ pin_subpackage("binutils_impl_" ~ ctng_target_platform, max_pin=None) }}
         # libstdcxx-ng is a runtime dep of gcc because LTO requires it.
         - {{ pin_subpackage("libstdcxx-ng", max_pin=None) }}
         - {{ pin_subpackage("libgcc-ng", max_pin=None) }}
         - {{ pin_subpackage("libgomp", max_pin=None) }}
+        - sysroot_{{ ctng_target_platform }}
     run_exports:
       # impose this requirement across the build/host boundary
       strong:
         - {{ pin_subpackage("libgcc-ng", max_pin=None) }}
     test:
+      requires:
+        - sysroot_{{ ctng_target_platform }} {{ conda_glibc_ver }}
```

Comment:
I'm still not sure why this is needed.

Reply:
I think it is conda build. 

Here is an install of binutils in the miniforge3 docker containers

```
(base) clarence:~ beckermr$ docker run --rm -it condaforge/miniforge3
(base) root@9926e2b6839a:/# conda install binutils
Collecting package metadata (current_repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: /opt/conda

  added / updated specs:
    - binutils


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    binutils-2.34              |       h53a641e_4          26 KB  conda-forge
    binutils_impl_linux-64-2.34|       h53a641e_4         9.2 MB  conda-forge
    ca-certificates-2020.4.5.1 |       hecc5488_0         146 KB  conda-forge
    certifi-2020.4.5.1         |   py37hc8dfbb8_0         151 KB  conda-forge
    conda-4.8.3                |   py37hc8dfbb8_1         3.0 MB  conda-forge
    ld_impl_linux-64-2.34      |       h53a641e_4         616 KB  conda-forge
    openssl-1.1.1g             |       h516909a_0         2.1 MB  conda-forge
    python_abi-3.7             |          1_cp37m           4 KB  conda-forge
    sysroot_linux-64-2.12      |       h3a0023d_5        30.7 MB  conda-forge
    ------------------------------------------------------------
                                           Total:        46.0 MB

The following NEW packages will be INSTALLED:

  binutils           conda-forge/linux-64::binutils-2.34-h53a641e_4
  binutils_impl_lin~ conda-forge/linux-64::binutils_impl_linux-64-2.34-h53a641e_4
  python_abi         conda-forge/linux-64::python_abi-3.7-1_cp37m
  sysroot_linux-64   conda-forge/noarch::sysroot_linux-64-2.12-h3a0023d_5

The following packages will be UPDATED:

  ca-certificates                     2019.11.28-hecc5488_0 --> 2020.4.5.1-hecc5488_0
  certifi                                 2019.11.28-py37_0 --> 2020.4.5.1-py37hc8dfbb8_0
  conda                                        4.8.2-py37_0 --> 4.8.3-py37hc8dfbb8_1
  ld_impl_linux-64                        2.33.1-h53a641e_8 --> 2.34-h53a641e_4
  openssl                                 1.1.1d-h516909a_0 --> 1.1.1g-h516909a_0


Proceed ([y]/n)? 
```

Note that it correctly grabs the 2.12 sysroot. 

Also, if I request the sysroot directly, I get

```
base) root@9926e2b6839a:/# conda install sysroot_linux-64
Collecting package metadata (current_repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: /opt/conda

  added / updated specs:
    - sysroot_linux-64


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    _sysroot_linux-64_curr_repodata_hack-3|       ha9d2b57_4           5 KB  conda-forge
    ca-certificates-2020.4.5.1 |       hecc5488_0         146 KB  conda-forge
    certifi-2020.4.5.1         |   py37hc8dfbb8_0         151 KB  conda-forge
    conda-4.8.3                |   py37hc8dfbb8_1         3.0 MB  conda-forge
    kernel-headers_linux-64-3.10.0|       h3a0023d_4         945 KB  conda-forge
    openssl-1.1.1g             |       h516909a_0         2.1 MB  conda-forge
    python_abi-3.7             |          1_cp37m           4 KB  conda-forge
    sysroot_linux-64-2.17      |       h3a0023d_4        32.5 MB  conda-forge
    ------------------------------------------------------------
                                           Total:        38.9 MB

The following NEW packages will be INSTALLED:

  _sysroot_linux-64~ conda-forge/noarch::_sysroot_linux-64_curr_repodata_hack-3-ha9d2b57_4
  kernel-headers_li~ conda-forge/noarch::kernel-headers_linux-64-3.10.0-h3a0023d_4
  python_abi         conda-forge/linux-64::python_abi-3.7-1_cp37m
  sysroot_linux-64   conda-forge/noarch::sysroot_linux-64-2.17-h3a0023d_4

The following packages will be UPDATED:

  ca-certificates                     2019.11.28-hecc5488_0 --> 2020.4.5.1-hecc5488_0
  certifi                                 2019.11.28-py37_0 --> 2020.4.5.1-py37hc8dfbb8_0
  conda                                        4.8.2-py37_0 --> 4.8.3-py37hc8dfbb8_1
  openssl                                 1.1.1d-h516909a_0 --> 1.1.1g-h516909a_0


Proceed ([y]/n)? 
```

In conda-build in the CI, I am seeing it grab 2.17. I think conda build is directly installing the deps of the things it builds instead of installing the built thing and letting the solver install the deps. This would explain the behavior we see and is consistent with the idea that version is higher than features when packages are requested directly.

Reply:
The docker container here is cos6 and I was seeing errors when building with the 2.17 sysroot related to the GLIBC version in the resulting packages at link time.

Reply:
Thanks. I'll have a look tomorrow and merge.